src/approaches/reinforcement_learning_approach.py

```diff
@@ -0,0 +1,109 @@
+    """A bilevel planning approach that learns NSRTs."""
+
+    def __init__(self, initial_predicates: Set[Predicate],
+                 initial_options: Set[ParameterizedOption], types: Set[Type],
+                 action_space: Box, train_tasks: List[Task]) -> None:
+        super().__init__(initial_predicates, initial_options, types,
+                         action_space, train_tasks)
+        self._nsrts: Set[NSRT] = set()
+        self.online_learning_cycle = 0
+        self.initial_dataset: List[LowLevelTrajectory] = []
+        self.online_dataset: List[LowLevelTrajectory] = []
+        # initialize option learner?
+
+    def _learn_pruned_nsrts(self,
+                            initial_trajectories: List[LowLevelTrajectory],
+                            trajectories: List[LowLevelTrajectory],
+                            online_learning_cycle: Optional[int]) -> None:
+        self._nsrts = learn_pruned_nsrts_from_data(
+            initial_trajectories,
+            trajectories,
+            self._train_tasks,
+            self._get_current_predicates(),
+            self._action_space,
+            sampler_learner=CFG.sampler_learner)
+        save_path = utils.get_approach_save_path_str()
+        with open(f"{save_path}_{online_learning_cycle}.NSRTs", "wb") as f:
+            pkl.dump(self._nsrts, f)
+
+    @classmethod
+    def get_name(cls) -> str:
+        return "reinforcement_learning"
+
+    def learn_from_offline_dataset(self, dataset: Dataset) -> None:
+        # The only thing we need to do here is learn NSRTs,
+        # which we split off into a different function in case
+        # subclasses want to make use of it.
+        self.initial_dataset = dataset.trajectories
+        self._learn_nsrts(dataset.trajectories, online_learning_cycle=None)
+
+    @classmethod
+    def _make_termination_fn(cls, goal: Set[GroundAtom]) \
+            -> Callable[[State], bool]:
+
+        def _termination_fn(s: State) -> bool:
+            return all(goal_atom.holds(s) for goal_atom in goal)
+
+        return _termination_fn
+
+    def get_interaction_requests(self) -> List[InteractionRequest]:
+        requests = []
+        for i in range(len(self._train_tasks)):
+            task = self._train_tasks[i]
+            try:
+                _act_policy = self.solve(task, CFG.timeout)
+            except (ApproachTimeout, ApproachFailure) as e:
+                partial_refinements = e.info.get("partial_refinements")
+                assert partial_refinements is not None
+                _, plan = max(partial_refinements, key=lambda x: len(x[1]))
+                _act_policy = utils.option_plan_to_policy(plan)
+            request = InteractionRequest(
+                train_task_idx=i,
+                act_policy=_act_policy,
+                query_policy=lambda s: None,
+                termination_function=ReinforcementLearningApproach.
+                _make_termination_fn(task.goal)  # pylint: disable=line-too-long
```

Comment:
you can do `self._make_termination_fn` and then get rid of the pylint disable

Reply:
checker complains that 'self' is unused which is why I did this. But maybe I can add some ignore statement so we can do that.

Reply:
to clarify, the checker complains if you leave the method definition alone, and just change this line?